Support of is_net_port and is_host_port externs.

komaljai opened the pull request.

psivanup reviewed (approved):
Looks fine

syogender commented on backends/tc/ebpfCodeGen.cpp:
```diff
@@ -1987,14 +1987,29 @@ void ControlBodyTranslatorPNA::processMethod(const P4::ExternMethod *method) {
+            if (methodCallExpr->method->toString() == "is_net_port") {
+                builder->append("bpf_p4tc_is_net_port(skb, ");
+            } else {
+                builder->append("bpf_p4tc_is_net_port(skb, ");
```
both adding ""bpf_p4tc_is_net_port(skb," ?

why check?
Testcases to exercise this code with 0 and argument and more than 1 arguments

komaljai replied:
It is incorrect code, it should be ""bpf_p4tc_is_host_port(skb,. I have fixed it.

I will add testcase for more than 1 argument.

vbnogueira commented on testdata/p4tc_samples/is_host_port.p4:
```diff
@@ -0,0 +1,128 @@
+        drop_packet();
+    }
+
+    @tc_acl("RUS:RXP") table nh_table {
```
Minor nit - since `nh_table` needs an entry to redirect, we should have create permissions
So maybe just remove `tc_acl` from here and keep the default permissions

```suggestion
    table nh_table {
```

vbnogueira commented on testdata/p4tc_samples/is_net_port.p4:
```diff
@@ -0,0 +1,128 @@
+        drop_packet();
+    }
+
+    @tc_acl("RUS:RXP") table nh_table {
```
Same as in the previous comment (https://github.com/komaljai/p4c/pull/28#discussion_r1709720040)

```suggestion
    table nh_table {
```

vbnogueira commented on testdata/p4tc_samples_outputs/is_host_port_control_blocks.c:
```diff
@@ -0,0 +1,306 @@
+        {
+if (/* hdr->ipv4.isValid() */
+            hdr->ipv4.ebpf_valid) {
+                bpf_p4tc_is_host_port(skb, skb->ifindex);
```
Also small detail, the return of `bpf_p4tc_is_host_port` is being lost

```suggestion
                tmp_0 = bpf_p4tc_is_host_port(skb, skb->ifindex);
```

komaljai replied:
addressed

AdarshRawat1 committed:
Docs : Content Organization and Documentation Updates (#4850)

* Create Subsections of related topics

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove Revision History

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Exclude Subsections from generated Documentation

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove: deprecated Section from generated docs

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Disable Todo list

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Docs : Hide Namespace and Classes section from output

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Minor Fix : missing "`" in directory structure

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove Revision History

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Disable Auto Link support for classes/namespaces

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge "Common P4C Utility Functions" as a subsection of "P4C" Page

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Convert file descriptions to Markdown table format in BMv2 "pna_nic" Backend

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove Revision History

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove Revision History

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Content for P4test Backend

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Refactor : Organize `portable_common` directory file descriptions

Organize `portable_common` directory file descriptions into a table format
Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Hide Doxygen Commands from GitHub Markdown render

Using the New Doxygen comment style in v1.12.0 i.e.
`<!-- ... -->`
For details - https://github.com/p4lang/p4c/pull/4861

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Create Subsections of related topics

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Docs : Hide Namespace and Classes section from output

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove Revision History

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C Overview with P4C root README

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Content for P4test Backend

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Fix : removed extra Command

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Organize `IR` Classes descriptions into a table format

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* chore: Add TOC to content-rich pages

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Remove TOC from Changelogs

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Exclude GitHub markdown TOC from Doxygen output

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Highlight sample backends in P4C with a dedicated heading

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merged: BMv2 "pna_nic" Backend into Behavioral Model Backend

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge P4C IR Classes into IR Page

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Change the top of tree view from "P4C" -> "P4 Compiler Documentation (P4C)"

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Change root page heading "P4C" -> "Getting Started"

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Move changelogs to the end of Side bar

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Improve Getting started page heading

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Merge "portable_common" section under "bmv2"

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Minor formatting Fixes

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Minor command fix

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Fix: Broken Table structure in Doxygen output

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Refactor : Page headings

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Refactor : Ordering of backends

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Enable Nice URL for generated docs

Current default doxygen URL config Output-
https://p4lang.github.io/p4c/md__2home_2runner_2work_2p4c_2p4c_2backends_2dpdk_2_r_e_a_d_m_e.html

Updated -
https://p4lang.github.io/p4c/dpdk_backend.html

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Doc: Add inclusion notes to Documented Markdown files

- Added comments to indicate whether each README is included as a standalone page, a subsection, or not included in the P4 compiler documentation.
- Provided specific links to relevant sections or pages within the documentation.

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Docs: Add remaining documentation inclusion notes

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Add GitHub profile links for Core developers of p4tools

Compatible with GitHub and Doxygen

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Fix: formatting of inclusion note

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Docs: Side bar grouping of backends

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Add Introduction for Frontend and Midend

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

* Update lib/README.md

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

Co-authored-by: Fabian Ruffy <5960321+fruffy@users.noreply.github.com>
Signed-off-by: Adarsh Rawat <adarshbunny293@gmail.com>

* Revert: Changes that are migrated to PR #4876

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>

---------

Signed-off-by: Adarsh <Adarshbunny293@gmail.com>
Signed-off-by: Adarsh Rawat <adarshbunny293@gmail.com>
Co-authored-by: Fabian Ruffy <5960321+fruffy@users.noreply.github.com>

komaljai committed:
Support of is_net_port and is_host_port externs.

Signed-off-by: Komal, Jain <komal.jain@intel.com>
